### PR TITLE
Binary_Value_Encode_Values_List, pre-requisite to COV subscribtions, …

### PR DESCRIPTION
**Pull request by metelik**
…function added

### CODE REVIEW

File: src/bacnet/basic/object/bv.c [hunk 1]
```diff
@@ -347,6 +347,58 @@ static int Binary_Value_Priority_Array_Encode(
...
+        bitstring_init(&value_list->value.type.Bit_String);
+        bitstring_set_bit(
+            &value_list->value.type.Bit_String, STATUS_FLAG_IN_ALARM, false);
+        bitstring_set_bit(
+            &value_list->value.type.Bit_String, STATUS_FLAG_FAULT, false);
+        bitstring_set_bit(
+            &value_list->value.type.Bit_String, STATUS_FLAG_OVERRIDDEN, false);
+        if (Binary_Input_Out_Of_Service(object_instance)) {
+            bitstring_set_bit(&value_list->value.type.Bit_String,
+                STATUS_FLAG_OUT_OF_SERVICE, true);
```
Comment (PeterJeaiem) [inline src/bacnet/basic/object/bv.c:389]: considered returning status = false when out of service ?

**Review by PeterJeaiem — APPROVED** [review]

**Review by alanj853 — APPROVED** [review]

File: src/bacnet/basic/object/bv.c [hunk 2]
```diff
@@ -347,6 +347,58 @@ static int Binary_Value_Priority_Array_Encode(
...
+        value_list->propertyArrayIndex = BACNET_ARRAY_ALL;
+        value_list->value.context_specific = false;
+        value_list->value.tag = BACNET_APPLICATION_TAG_ENUMERATED;
+        value_list->value.next = NULL;
+        value_list->value.type.Enumerated =
+            Binary_Value_Present_Value(object_instance);
+        value_list->priority = BACNET_NO_PRIORITY;
+        value_list = value_list->next;
+    }
+    if (value_list) {
```
[the remark this answers is not in the document]
Comment (PeterJeaiem) [reply src/bacnet/basic/object/bv.c:374]: was my first reaction too, but we've moved to the next on line 372, and we're checking that

[the remark this answers is not in the document]
Comment (metelik) [reply src/bacnet/basic/object/bv.c:374]: see line 372 ;) this is merely adopt and modify from very akin Binary_Input unit...

File: src/bacnet/basic/object/bv.c (hunk 1 above)
Comment (metelik) [reply src/bacnet/basic/object/bv.c:389]: I took it from:
https://github.com/se-apc/bacnet-stack/blob/19ded72ff2592d8142f139626c1b9d5386d8a016/src/bacnet/basic/object/bi.c#L283
and in lets say any others be it analogue value or input the OOS is not taken into account for the status of encoding.... I infer this is how it is supposed to be...

**Pull request merged**